joblib/_store_backends.py

```diff
@@ -30,9 +31,10 @@ class CacheWarning(Warning):
 
 def concurrency_safe_write(object_to_write, filename, write_func):
     """Writes an object into a unique file in a concurrency-safe way."""
+    hostname = socket.getfqdn()
     thread_id = id(threading.current_thread())
-    temporary_filename = '{}.thread-{}-pid-{}'.format(
-        filename, thread_id, os.getpid())
+    temporary_filename = '{}.hostname-{hostname}-thread-{}-pid-{}'.format(
```

Comment:
Either move to a f-string or don't put name for consistency.
```suggestion
    temporary_filename = '{}.hostname-{}-thread-{}-pid-{}'.format(
```